manual-provision-payg/testsuite/code_gen.py

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/env python
+import dbus
+import sys
+import os
+from os.path import expanduser
+import string
+import random
+import textwrap
+import json
+import os.path
+import hashlib
+import textwrap
+import struct
+import hmac
+import sqlite3
+import csv
+import errno
+from string import Template
+
+key_code_file_name = "paygo-test-codes"
+
+device_id = ""
+
+# available_times = ["5s", "1m", "5m", "30m", "1h", "8h", "1d", "2d", "3d", "4d", "5d", "6d", "7d",
+#                    "8d", "9d", "10d", "11d", "12d", "13d", "14d", "30d", "60d", "90d", "120d", "365d", "infinite"]
+
+paygo_enabled_rules = """[PAYG]
+Enabled=true"""
+
+period = 3
+
+print("\n")
+
+cwd = ""
+
+
+def set_usb_path():
+    for drive in enumerate_drives():
+        if get_usb_path(drive):
+            global cwd
+            cwd = get_usb_path(drive)
```

Comment:
`get_usb_path(device)` filters the first drive which contains "Endless_Factory_Test.tar" in it's root directory. This should guarantee that the files are exported to the correct drive. ie
` if os.path.isfile(path + "Endless_Factory_Test.tar"):
                usb_file_path = path`

Unless ofcourse twodrives meet this condition in which case it will select the first.